### PR TITLE
switch to pure async factory pattern

### PR DESCRIPTION
**Pull request by juliangruber**
Instead of also offering the internal queue, switch to a pure async factory pattern and simplify flow control in general.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -85,17 +79,22 @@ function MultiDrive (name, opts, cb) {
       series(fns, function (err) {
         if (err) return done(err)
         var drive = hyperdrive(db)
-        var _opts = (secretKey)
-          ? xtend(self.opts, opts, { secretKey: secretKey })
-          : xtend(self.opts, opts)
-        var archive = drive.createArchive(key, _opts)
-        self.archives[name] = archive
-        done()
+        var __opts = (secretKey)
```
Comment (yoshuawuyts) [inline index.js]: redeclaring of `_opts` is a bit confusing - would've assumed `standard` would throw actually; maybs remove `var` here?

File: index.js [hunk 2]
```diff
@@ -27,28 +26,23 @@ function MultiDrive (name, opts, cb) {
...
-    cb(null, self)
+    if (err) cb(err)
   })
 
   function sink (pairs) {
-    mapLimit(pairs, 1, iterator, function (err) {
+    mapLimit(pairs, 1, iterator, function (err, archivesArray) {
       if (err) return cb(err)
-      while (self.queue.length) self.queue.shift()()
+      var archivesMap = keyBy(archivesArray, archive => archive.metadata.directory)
```
Comment (yoshuawuyts) [inline index.js]: given this is library code that's meant to be compatible with `hyperdrive` and `hypercore`, could we keep this completely ES5? - thanks!

File: index.js [hunk 3]
```diff
@@ -27,28 +26,23 @@ function MultiDrive (name, opts, cb) {
...
-
-  this.db = namedLevel(name)
-  this.opts = opts
-  this.archives = {}
-  this.queue = []
+  var db = namedLevel(name)
 
-  var rs = this.db.createReadStream()
+  var rs = db.createReadStream()
   pump(rs, concat({ encoding: 'json' }, sink), (err) => {
```
Comment (yoshuawuyts) [inline index.js]: oops, missed this one myself haha

**Comment by yoshuawuyts** [comment]
nice, digging this :sparkles: :grin:

File: index.js (hunk 2 above)
Comment (juliangruber) [inline index.js]: i'll clean it up

File: index.js (hunk 1 above)
Comment (juliangruber) [inline index.js]: it's underscore underscore opts here, a bit hard to read. what about this new commit?

**Comment by juliangruber** [comment]
@yoshuawuyts mind taking another look? :)

**Comment by yoshuawuyts** [comment]
yeah I think this looks goo :grin:

**Comment by yoshuawuyts** [comment]
Wooh! ✨

**Pull request merged**